src/CoreWCF.WebHttp/src/CoreWCF/OpenApi/Middleware/OpenApiMiddleware.cs

```diff
@@ -0,0 +1,106 @@
+                { "/swagger/index.html", ReadResource("index.html") },
+                { "/swagger/swagger-ui-bundle.js", ReadResource("swagger-ui-bundle.js") },
+                { "/swagger/swagger-ui-standalone-preset.js", ReadResource("swagger-ui-standalone-preset.js") },
+                { "/swagger/swagger-ui.css", ReadResource("swagger-ui.css") },
+            };
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            string requestPath = context.Request.Path.ToString();
+            if (!requestPath.StartsWith("/"))
+            {
+                requestPath = "/" + requestPath;
+            }
+
+            // Needs to happen after contracts have been added to document provider.
+            if (OpenApiDocumentationEnabled() && requestPath == "/swagger/swagger.json")
+            {
+                await WriteResponse(context, requestPath, Encoding.UTF8.GetBytes(_openApiDocumentProvider.Document));
+            }
+            else if (OpenApiDocumentationEnabled() && _swaggerUi.ContainsKey(requestPath))
+            {
+                await WriteResponse(context, requestPath, _swaggerUi[requestPath]);
+            }
+            else
+            {
+                await _next.Invoke(context).ConfigureAwait(false);
+            }
+        }
+
+        private bool OpenApiDocumentationEnabled() => _openApiDocumentProvider.Contracts.Count > 0;
+
+        private static string GetContentType(string path)
+        {
+            switch (Path.GetExtension(path))
+            {
+                case ".png":
+                    return "image/png";
+                case ".html":
+                    return "text/html";
+                case ".js":
+                    return "text/javascript";
+                case ".json":
+                    return "";
+                case ".css":
+                    return "text/css";
+                default:
+                    return null;
+            }
+        }
+
+        private async Task WriteResponse(HttpContext context, string requestPath, byte[] response)
+        {
+            context.Response.StatusCode = 200;
+            context.Response.ContentType = GetContentType(requestPath);
+            await context.Response.Body.WriteAsync(response, 0, response.Length);
+        }
+
+        private byte[] ReadResource(string filename)
+        {
+            Assembly assembly = typeof(OpenApiMiddleware).Assembly;
+
+            string resourceName = assembly.GetManifestResourceNames()
+                .First(resourceName => resourceName.EndsWith(filename, StringComparison.Ordinal));
+
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            using BinaryReader reader = new BinaryReader(stream);
+            return reader.ReadBytes((int)stream.Length);
+        }
```

Comment:
It's not super obvious that the return statement is part of the using blocks. Can you wrap the return in braces so it's obvious.

Reply:
This is addressed.